src/plugins/btrfs.c

```diff
@@ -454,75 +460,37 @@ gboolean bd_btrfs_delete_subvolume (const gchar *mountpoint, const gchar *name,
  * Tech category: %BD_BTRFS_TECH_SUBVOL-%BD_BTRFS_TECH_MODE_QUERY
  */
 guint64 bd_btrfs_get_default_subvolume_id (const gchar *mountpoint, GError **error) {
-    GRegex *regex = NULL;
-    GMatchInfo *match_info = NULL;
-    gboolean success = FALSE;
-    gchar *output = NULL;
-    gchar *match = NULL;
+    enum btrfs_util_error err;
     guint64 ret = 0;
-    const gchar *argv[5] = {"btrfs", "subvol", "get-default", mountpoint, NULL};
 
-    if (!check_deps (&avail_deps, DEPS_BTRFS_MASK, deps, DEPS_LAST, &deps_check_lock, error) ||
-        !check_module_deps (&avail_module_deps, MODULE_DEPS_BTRFS_MASK, module_deps, MODULE_DEPS_LAST, &deps_check_lock, error))
-        return 0;
-
-    regex = g_regex_new ("ID (\\d+) .*", 0, 0, error);
-    if (!regex) {
-        bd_utils_log_format (BD_UTILS_LOG_WARNING, "Failed to create new GRegex");
-        /* error is already populated */
-        return 0;
+    err = btrfs_util_get_default_subvolume (mountpoint, &ret);
+    if (err) {
+        g_set_error (error, BD_BTRFS_ERROR, BD_BTRFS_ERROR_NOT_FOUND, format_btrfs_error (err));
     }
 
-    success = bd_utils_exec_and_capture_output (argv, NULL, &output, error);
-    if (!success) {
-        g_regex_unref (regex);
-        return 0;
-    }
-
-    success = g_regex_match (regex, output, 0, &match_info);
-    if (!success) {
-        g_set_error (error, BD_BTRFS_ERROR, BD_BTRFS_ERROR_PARSE, "Failed to parse subvolume's ID");
-        g_regex_unref (regex);
-        g_match_info_free (match_info);
-        g_free (output);
-        return 0;
-    }
-
-    match = g_match_info_fetch (match_info, 1);
-    ret = g_ascii_strtoull (match, NULL, 0);
-
-    g_free (match);
-    g_match_info_free (match_info);
-    g_regex_unref (regex);
-    g_free (output);
-
     return ret;
 }
 
 /**
  * bd_btrfs_set_default_subvolume:
  * @mountpoint: mountpoint of the volume to set the default subvolume ID of
  * @subvol_id: ID of the subvolume to be set as the default subvolume
- * @extra: (nullable) (array zero-terminated=1): extra options for the setting (right now
- *                                                 passed to the 'btrfs' utility)
  * @error: (out) (optional): place to store error (if any)
  *
  * Returns: whether the @mountpoint volume's default subvolume was correctly set
  * to @subvol_id or not
  *
  * Tech category: %BD_BTRFS_TECH_SUBVOL-%BD_BTRFS_TECH_MODE_MODIFY
  */
-gboolean bd_btrfs_set_default_subvolume (const gchar *mountpoint, guint64 subvol_id, const BDExtraArg **extra, GError **error) {
-    const gchar *argv[6] = {"btrfs", "subvol", "set-default", NULL, mountpoint, NULL};
```

Comment:
This breaks [udisks](https://github.com/storaged-project/libblockdev/actions/runs/7616878904/job/20744641672?pr=999), which I think we rather not? But -Werror will scream at me if I keep it around. I guess I should mark it with something __unused?

Reply:
Yes, existing public API cannot be changed. Please add `G_GNUC_UNUSED` annotation.

If you want to keep full API compatibility, you'd need to translate any known extra args to library calls/arguments. Porting existing API is tough, creating new API from scratch is much easier.

Reply:
I think deprecating the argument would be the best, I am not sure what extra options can be given as `btrfs subvolume set default` only takes:

```
btrfs subvolume set-default <subvolid> <path>
```

So I think for backwards compatibility this would actually work out quite well in this instance. Should I add `G_DEPRECATED` as well to the argument and update the comment?

Reply:
No need to special annotation for function arguments I guess.